src/couch_replicator/src/couch_replicator_changes_reader.erl

```diff
@@ -168,14 +168,14 @@ process_change({last_seq, _}, _) ->
     ok.
 
 is_doc_id_too_long(IdLength) ->
-    ConfigDocIdLength = config:get("replicator", "max_document_id_length", ?DOC_ID_LIMIT_BYTES),
+    ConfigDocIdLength = config:get_integer("replicator", "max_document_id_length", ?DOC_ID_LIMIT_BYTES),
     case ConfigDocIdLength > ?DOC_ID_LIMIT_BYTES of
         true ->
             couch_log:error(
                 "Config value of ~p for max_document_id_length is greater than the limit: ~p ",
                 [ConfigDocIdLength, ?DOC_ID_LIMIT_BYTES]
             ),
-            ?DOC_ID_LIMIT_BYTES;
+            true;
         false ->
-            ConfigDocIdLength
+            false
```

Comment:
This is a tricky one. Replicator running on TXE could in theory replicate between two 3.x instance (target and source are on 3.x and replicator on TXE) and it should be able to replicate longer IDs than what TXE supports. So I think we should not update the replicator in this PR.

Extra background info: The reason this limit exists, from what I can recall is to prevent the replicator from propagating an abnormally long ID that could not be parsed in individual HTTP requests but could still be replicated with the _bulk_docs endpoint

Reply:
Good thought! I have removed the changes to the replicator for now. See https://github.com/apache/couchdb/pull/3632/commits/468123da973d22ff9d4e91477b1f48568cde80b9.